src/Components/Comment.js

```diff
@@ -1,63 +1,57 @@
 import axios from "axios";
 import { useEffect, useState } from "react";
-
-const apiUrlComment1 = "https://blogpostapi1.herokuapp.com/comment";
-const apiUrlComment2 = "https://blogpostapi1.herokuapp.com/comment/all_comments/";
-
+import { apiUrl } from "./NewBlog";
 
 const Comment = ({ blogPost }) => {
-  const [displayComment, setDisplayComment] = useState([]);
-  const [comment, setComment] = useState("");
+  const [comments, setComments] = useState([]);
+  const [content, setContent] = useState("");
 
   const handleClick = async (e) => {
     e.preventDefault();
     try {
-      await axios.post(apiUrlComment1, {
-        content: comment,
+      await axios.post(`${apiUrl}comment/`, {
+        content,
         post: blogPost.id,
       });
     } catch {
       console.log("Post Error");
     }
-    setComment("")
+    setContent("");
   };
 
-
   useEffect(() => {
     const getComment = async () => {
       try {
         const { data } = await axios.get(
-          apiUrlComment2 + blogPost.id
+          `${apiUrl}comment/all_comments/` + blogPost.id
         );
-        setDisplayComment(data.data);
+        setComments(data.data);
       } catch {
         console.log("FETCH COMMENT ERROR");
       }
     };
     getComment();
-  }, [handleClick]);
+  }, [comments]);
```

Comment:
This useEffect does not have any dependency.

If you are trying to refetch the list of comments after successfully adding a new comment, you would need another state variable to track that

Reply:
Okay sir